src/Raven.Server/Documents/Includes/IncludeCompareExchangeValuesCommand.cs

```diff
@@ -63,26 +64,36 @@ internal void Gather(Document document)
             if (_includes == null || _includes.Length == 0)
                 return;
 
-            if (_includedKeys == null)
-                _includedKeys = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            _includedKeys ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
             foreach (var include in _includes)
                 IncludeUtil.GetDocIdFromInclude(document.Data, new StringSegment(include), _includedKeys, _database.IdentityPartsSeparator);
         }
 
-        internal void Materialize()
+        public void AddDocument(string id)
+        {
+            _includedKeys ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            _includedKeys.Add(id);
+        }
+
+        public long? GetAtomicGuardIndex(string key, long maxAllowedRaftId)
+        {
+            if (_serverContext == null) CreateServerContext();
+            var value = _database.ServerStore.Cluster.GetCompareExchangeValue(_serverContext, CompareExchangeKey.GetStorageKey(_database.Name, key));
+
+            if(value.Index > maxAllowedRaftId)
+                return null; // we are seeing partially committed value, skip it
+            if (value.Index < 0)
```

Comment:
when can this happened?

Reply:
That can happen if the atomic guard does not exists